### PR TITLE
[Feature 24] Entity Tests and Use Case Updates

### PR DESCRIPTION
**Pull request by harshsbajwa**
# Description

1.  removed HatTest, created PetTest, ToDoListTest, DoneListTest, modified CustomizationTest, ItemTest, made minor revisions to DoneList, ToDoList, Pet, Item (removed ImageIcon), and documentation changes. Updated BuyItemUC and BuyItemUCTest.

2.  Made changes to AdoptPetUC, BuyHeartUC, PetShopUIController, and documentation changes to UserUC. Added AdoptPetInputBoundary, BuyHeartController, CheckerUC, PetOutputBoundary, PetRefresher, and TasklistRefresher

Fixes # (issue)

## Type of change

- [ ] Compilation error fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [x] Test cases
- [x] Redundant code removal (fix code smells)
- [ ] Data storage 
- [ ] UI element

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/useCases/CheckerUC.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+     * Checker Use case interactor.
+     * When the user opens the program this will check if any deadlines are past their due date and remove health
+     * from the pet accordingly.
+     */
+
+    public void checkDeadlines(){
+
+    }
+
+}
```
Comment (XinyingXu) [inline src/main/java/useCases/CheckerUC.java:14]: You need to resolve merge conflicts because we have made changes to ToDoList, DoneList and CheckerUC.

Comment (harshsbajwa) [reply src/main/java/useCases/CheckerUC.java:51]: Thanks for pointing this out, I resolved the conflicts

**Review by XinyingXu — APPROVED** [review]
Conflicts resolved and other parts of the code look good. Good job on documentation and tests.

**Review by BetterthanNothin — APPROVED** [review]
Looks Good. Proud of you son.

**Pull request merged**